### DISCUSSION TITLE
Output templates for GPT response

### DISCUSSION THREAD

**Discussion by Sinled** [chhoumann/quickadd#536] [Q&A]
Hi, is there possibility to have two templates for Ai assistant? One for input and another for output?

I want to receive JSON  response from Chat GPT, and place fields from this response in different places in output template.

It seems it is not possible currently or am i missing something?

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

Yes, this is possible now. The script API gained schema-constrained structured output: the model is forced to return JSON matching your schema, and you place each field wherever you want in a template - which is exactly the input/output split you described.

In a Macro, add a user script step like:

```js
module.exports = async (params) => {
    const { object } = await params.quickAddApi.ai.agent({ model: "gpt-5" }).generate({
        prompt: "Summarise this: {{VALUE:input}}",
        schema: {
            type: "object",
            properties: {
                title: { type: "string" },
                summary: { type: "string" },
                tags: { type: "array", items: { type: "string" } },
            },
            required: ["title", "summary", "tags"],
        },
    });
    if (!object) throw new Error("Model did not return valid JSON");

    params.variables.title = object.title;
    params.variables.summary = object.summary;
    params.variables.tags = object.tags.join(", ");
};
```

Then make the next macro step a Nested Choice that runs a Template choice - that template file is your output template, with `{{VALUE:title}}`, `{{VALUE:summary}}`, `{{VALUE:tags}}` placed wherever you like. (The `prompt` string is run through the normal format syntax, so it can pull in prompts, selection, clipboard, etc.)

The stored AI Assistant macro command on its own only gives you the whole response as a single variable, so per-field placement needs this small script.

Docs: https://quickadd.obsidian.guide/docs/QuickAddAPI - see the "Structured output" section.
